acceptance/tests/api-gateway/api_gateway_tenancy_test.go

```diff
@@ -0,0 +1,421 @@
+				require.EqualValues(r, "gateway", httproute.Status.Parents[0].ParentRef.Name)
+				require.NotNil(r, httproute.Status.Parents[0].ParentRef.Namespace)
+				require.EqualValues(r, gatewayNamespace, *httproute.Status.Parents[0].ParentRef.Namespace)
+				checkStatusCondition(r, httproute.Status.Parents[0].Conditions, trueCondition("Accepted", "Accepted"))
+				checkStatusCondition(r, httproute.Status.Parents[0].Conditions, trueCondition("ResolvedRefs", "ResolvedRefs"))
+			})
+
+			// now check to make sure that the route is actually created
+			retryCheck(t, 2, func(r *retry.R) {
+				// since we're not bound, check to make sure that the route doesn't target the gateway in Consul.
+				entry, _, err := consulClient.ConfigEntries().Get(api.HTTPRoute, "route", &api.QueryOptions{
+					Namespace: routeNamespace,
+				})
+				require.NoError(r, err)
+				route := entry.(*api.HTTPRouteConfigEntry)
+
+				require.EqualValues(r, "route", route.Meta["k8s-name"])
+				require.EqualValues(r, routeNamespace, route.Meta["k8s-namespace"])
+				require.Len(r, route.Parents, 1)
+			})
+
+			// and check to make sure that the certificate exists
+			retryCheck(t, 2, func(r *retry.R) {
+				entry, _, err := consulClient.ConfigEntries().Get(api.InlineCertificate, "certificate", &api.QueryOptions{
+					Namespace: certificateNamespace,
+				})
+				require.NoError(r, err)
+				certificate := entry.(*api.InlineCertificateConfigEntry)
+
+				require.EqualValues(r, "certificate", certificate.Meta["k8s-name"])
+				require.EqualValues(r, certificateNamespace, certificate.Meta["k8s-namespace"])
+			})
+		})
+	}
+}
+
+func applyFixture(t *testing.T, cfg *config.TestConfig, k8sOptions *terratestk8s.KubectlOptions, fixture string) {
+	t.Helper()
+
+	out, err := k8s.RunKubectlAndGetOutputE(t, k8sOptions, "apply", "-k", path.Join("../fixtures", fixture))
+	require.NoError(t, err, out)
+	helpers.Cleanup(t, cfg.NoCleanupOnFailure, func() {
+		k8s.RunKubectlAndGetOutputE(t, k8sOptions, "delete", "-k", path.Join("../fixtures", fixture))
+	})
+}
+
+func createNamespace(t *testing.T, ctx environment.TestContext, cfg *config.TestConfig) (string, *terratestk8s.KubectlOptions) {
+	t.Helper()
+
+	namespace := helpers.RandomName()
+
+	logger.Logf(t, "creating Kubernetes namespace %s", namespace)
+	k8s.RunKubectl(t, ctx.KubectlOptions(t), "create", "ns", namespace)
+	helpers.Cleanup(t, cfg.NoCleanupOnFailure, func() {
+		k8s.RunKubectl(t, ctx.KubectlOptions(t), "delete", "ns", namespace)
+	})
+
+	return namespace, &terratestk8s.KubectlOptions{
+		ContextName: ctx.KubectlOptions(t).ContextName,
+		ConfigPath:  ctx.KubectlOptions(t).ConfigPath,
+		Namespace:   namespace,
+	}
+}
+
+type certificateInfo struct {
+	Cert          *x509.Certificate
+	PrivateKey    *rsa.PrivateKey
+	CertPEM       []byte
+	PrivateKeyPEM []byte
+}
+
+func generateCertificate(t *testing.T, ca *certificateInfo, commonName string) *certificateInfo {
```

Comment:
If this is generalizable enough, it might be better to have it in the "framework" so it can be re-used.

Reply:
yeah, I'm considering it, but I do have the feeling that given the fact that we're the only ones who really do any sort of "bring your own cert" -- this will probably just be something we use... so figure maybe we can wait until we see it needed other places

Reply:
Yeah, that doesn't need to be in this PR.